scripts/ci/build_and_upload_wheels.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+"""
+Build and upload wheels to GCS.
+
+Install dependencies:
+    python3 -m pip install google-cloud-storage==2.9.0
+
+Use the script:
+    python3 scripts/ci/release.py --help
+"""
+from __future__ import annotations
+
+import argparse
+import os
+import platform
+import shutil
+import subprocess
+import tempfile
+from enum import Enum
+
+from google.cloud.storage import Bucket
+from google.cloud.storage import Client as Gcs
+
+
+def run(
+    cmd: str,
+    *,
+    cwd: str | None = None,
+    env: dict[str, str] | None = None,
+) -> None:
+    print(f"{cwd or ''}> {cmd}")
+    subprocess.check_output(cmd.split(), cwd=cwd, env=env)
+
+
+def detect_target() -> str:
+    arch = platform.machine()
+    if arch == "x86_64" or arch == "aarch64":
+        pass  # leave it as is
+    elif arch == "arm64":
+        arch = "aarch64"
+    else:
+        raise Exception(f"unknown architecture: {arch}")
+
+    os = platform.system()
+    if os == "Linux":
+        os = "unknown-linux-gnu"
+    elif os == "Darwin":
+        os = "apple-darwin"
+    elif os == "Windows":
+        os = "pc-windows-msvc"
+    else:
+        raise Exception(f"unknown platform: {os}")
+
+    return f"{arch}-{os}"
+
+
+def detect_pixi() -> bool:
+    path = os.environ.get("PATH")
+    return path is not None and ".pixi/env/bin" in path
+
+
+class BuildMode(Enum):
+    PYPI = "pypi"
+    PR = "pr"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+def build_and_upload(
+    bucket: Bucket,
+    mode: BuildMode,
+    gcs_dir: str,
+    target: str,
+) -> None:
+    if detect_pixi():
+        raise Exception("the build script cannot be started in the pixi environment")
+
+    if mode is BuildMode.PYPI:
+        # Only build web viewer when publishing to pypi
+        run("cargo run --locked -p re_build_web_viewer -- --release")
+        maturin_feature_flags = "--no-default-features --features pypi"
+    elif mode is BuildMode.PR:
+        maturin_feature_flags = "--no-default-features --features extension-module"
+
+    dist = f"dist/{target}"
+
+    # Build into `dist`
+    run(
+        "maturin build "
+        # "--compatibility manylinux_2_31 "
+        "--manifest-path rerun_py/Cargo.toml "
+        "--release "
+        f"--target {target} "
+        f"{maturin_feature_flags} "
+        f"--out {dist}"
+    )
+
+    # Repack wheel
+    with tempfile.TemporaryDirectory() as tmpdir:
+        print("Repacking wheel…")
+        pkg = os.listdir(dist)[0]
+        run(f"wheel unpack {dist}/{pkg} --dest {tmpdir}")
+        pkg_folder = os.listdir(tmpdir)[0]
+        shutil.rmtree(dist, ignore_errors=True)
+        os.makedirs(dist, exist_ok=True)
+        run(f"wheel pack {tmpdir}/{pkg_folder} --dest {dist}")
+        pkg = os.listdir(dist)[0]
```

Comment:
I think this can go away all together. The only reason we needed to repack wheels before was to inject the built-in rrd after building, but now it looks like we're just unpacking and repacking the wheel without modification.

Reply:
ok, will remove